pkg/task/task.go

```diff
@@ -791,19 +794,42 @@ func (m *taskExecutor) deleteExtnV1B1Deployment() (err error) {
 	return
 }
 
+// getExtnV1B1ReplicaSet will get the Replicaset as specified in the RunTask
+func (m *taskExecutor) getExtnV1B1ReplicaSet() (err error) {
+	client := replicaset.KubeClient(
+		replicaset.WithNamespace(m.getTaskRunNamespace()))
+
+	rs, err := client.GetRaw(m.getTaskObjectName())
+	if err != nil {
+		return
+	}
+
+	util.SetNestedField(m.templateValues, rs, string(v1alpha1.CurrentJSONResultTLP))
+	return
+}
+
 // deleteExtnV1B1ReplicaSet will delete one or more ReplicaSets as specified in
 // the RunTask
 func (m *taskExecutor) deleteExtnV1B1ReplicaSet() (err error) {
 	objectNames := strings.Split(strings.TrimSpace(m.getTaskObjectName()), ",")
+	client := replicaset.KubeClient(
+		replicaset.WithNamespace(m.getTaskRunNamespace()))
+
 	for _, name := range objectNames {
-		err = m.getK8sClient().DeleteExtnV1B1ReplicaSet(strings.TrimSpace(name))
+		err = client.Delete(strings.TrimSpace(name))
 		if err != nil {
 			return
 		}
 	}
 	return
 }
 
+func (m *taskExecutor) listExtnV1B1ReplicaSet(opt metav1.ListOptions) ([]byte, error) {
+	client := replicaset.KubeClient(
+		replicaset.WithNamespace(m.getTaskRunNamespace()))
+	return client.ListRaw(&opt)
```

Comment:
can we combine both of above with proper indentation

Reply:
done